aten/src/ATen/native/xnnpack/Activation.cpp

```diff
@@ -0,0 +1,64 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/native/utils/Factory.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+Tensor hardswish(const Tensor& input) {
+  using namespace internal;
+
+  const Tensor padded_input =
+      mobile::allocate_padded_contiguous_if_needed(
+          input, input.suggest_memory_format());
+
+  Tensor output = mobile::empty_with_tail_padding(
+      padded_input.sizes(),
+      padded_input.options().dtype(),
+      input.suggest_memory_format(),
+      padded_input.names());
+
+  xnn_operator_t hardswish_op{};
+  const auto channels = Layout::ActivationND::channel(padded_input.sizes());
```

Comment:
I wonder if you should just do padded_input.sizes().back()? It seems because we are not quite extracting channel. Or at least leave a comment, because it get confusing how sizes are stored in the same way between NCHW and NHWC.

Reply:
That is what ::channel does here. Use that function since we use ::batch as well and they work together well to get last dimension and all but last dimension from an ND tensor. We could rename them to indicate these are last dimension and rest but I think we're following xnnpack's definition here for "batch"

Reply:
I think "batch" is fine. My concern is mainly around channels. It is bit of a misnomer here. So maybe just rename that?